include/sys/zfs_rlock.h

```diff
@@ -45,7 +45,7 @@ struct locked_range;
 
 typedef void (rangelock_cb_t)(struct locked_range *, void *);
 
-typedef struct rangelock {
+typedef struct zfs_rangelock {
```

Comment:
I thought we already added a `zfs_` prefix to these types, but it seems we only updated the function prototypes.  What do you think about prefixing all of these types to prevent future collisions.

rangelock_type_t -> zfs_rangelock_type_t
rangelock_t -> zfs_rangelock_t
locked_range_t -> zfs_locked_range_t
rangelock_cb_t -> zfs_rangelock_cb_t

Reply:
OK